File: simulations/py/single_q.py
import random
import math

run = 10
mean = 10
sd = 1.5
mue = 9.5
sigma = 1.0
wt = 0
iat = 0
cat = 0
cwt = 0
cit = 0
se = 0

with open("output.txt", "w") as outfile:
    outfile.write("\ni r' IAT CAT SB r' ST SE WT IT \n")
    for j in range(1, run + 1):
        sum_x = sum(random.random() for _ in range(12))
        x1 = mean + sd * (sum_x - 6)
        iat = x1
        cat += iat

        if cat <= se:
            sb = se
            wt = se - cat
            cwt += wt
        else:
            sb = cat
            it = sb - se
            cit += it

        sum_x2 = sum(random.random() for _ in range(12))
        x2 = mue + sigma * (sum_x2 - 6)
        st = x2
        se = sb + st

        outfile.write(f"{j}\t{round(x1, 4)}\t{round(iat, 4)}\t{round(cat, 4)}\t{round(sb, 4)}"
                      f"\t{round(x2, 4)}\t{round(st, 4)}\t{round(se, 4)}\t{round(wt, 4)}"
                      f"\t{round(it, 4)}\n")

    awt = cwt / run
    pcu = (cat - cit) * 100 / cat
    outfile.write("Average waiting time\n")
    outfile.write(f"{awt}\n")
    outfile.write("Percentage capacity utilization\n")
    outfile.write(f"{pcu}\n")

File: simulations/js/single_q.js
const fs = require('fs');

function main() {
    let i, j, run = 10;
    let x, x1, x2, st, awt, pcu, wt = 0, iat = 0, it;
    let mean = 10, sd = 1.5, mue = 9.5, sigma = 1.0;
    let sb = 0, se = 0, cit = 0, cat = 0, cwt = 0;
    let outfile = fs.createWriteStream('output.txt', { flags: 'w' });
    outfile.write("\ni r' IAT CAT SB r' ST SE WT IT \n");
    for (j = 1; j <= run; j++) {
        
        let sum = 0;
        for (i = 1; i <= 12; i++) {
            x = Math.random();
            sum += x;
        }
        x1 = mean + sd * (sum - 6);
        iat = x1;
        
        cat += iat;
        
        if (cat <= se) {
            sb = se;
            wt = se - cat;
            cwt = cwt + wt;
            
        } else {
            sb = cat;
            it = sb - se;
            cit = cit + it;
        }
        
        sum = 0;
        for (i = 1; i <= 12; i++) {
            x = Math.random();
            sum += x;
        }
        x2 = mue + sigma * (sum - 6);
        st = x2;
        se = sb + st;
        
        outfile.write(`${j}\t${x1.toFixed(4)}\t${iat.toFixed(4)}\t${cat.toFixed(4)}\t${sb.toFixed(4)}\t${x2.toFixed(4)}\t${st.toFixed(4)}\t${se.toFixed(4)}\t${se.toFixed(4)}\t${wt.toFixed(4)}\t${it.toFixed(4)}\n`);
    }
    awt = cwt / run;
    pcu = (cat - cit) * 100 / cat;
    outfile.write("Average waiting time\n");
    outfile.write(`${awt}\n`);
    outfile.write("Percentage capacity utilization\n");
    outfile.write(`${pcu}\n`);
    outfile.end();
}

main();

